### PR TITLE
feat(PPDSC-2781): update menu defaults

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2781
As a user, I want the Menu component default styling to be updated as per the [designs](https://www.figma.com/file/Pthw7xMwBf3k9OoBC3HYWA/Component-updates?node-id=32%3A80008&t=WCxGyUDI6dPdb9DB-0), so that there is parity between Figma and code.

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2781-menu-defaults
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2781-menu-defaults/storybook

File: site/pages/components/menu.tsx [hunk 1]
```diff
@@ -819,12 +819,27 @@ const MenuComponent = (layoutProps: LayoutProps) => (
               default: 'space020',
               description: `Overrides the space between the leading icon and the label, as well as the label and the trailing icon`,
             },
-
             {
-              attribute: 'spaceInset',
```
Comment (mstuartf) [inline site/pages/components/menu.tsx:824]: Is this being removed because the prop is deprecated?

Comment (Xin00163) [reply site/pages/components/menu.tsx:824]: Yes. We can do it in https://nidigitalsolutions.jira.com/browse/PPDSC-2817 but I thought I might as well

File: src/menu/defaults.ts [hunk 2]
```diff
@@ -13,41 +34,85 @@ export default {
...
     horizontal: {
-      minHeight: 'sizing080',
-      stylePreset: 'menuItemHorizontal',
-      transitionPreset: 'backgroundColorChange',
-      spaceInline: 'space020',
-      spaceInset: 'spaceInsetSquish030',
+      small: {
+        ...smallCommonDefaults,
+        stylePreset: 'menuItemHorizontal',
+        transitionPreset: 'backgroundColorChange',
```
Comment (mutebg) [inline src/menu/defaults.ts:40]: the transition preset also seems to be the same across all variations, maybe move it to common defaults?

**Review by mutebg — APPROVED** [review]

**Review by baburay23 — APPROVED** [review]

**Pull request merged**